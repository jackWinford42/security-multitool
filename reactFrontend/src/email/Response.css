#progressBar {
    width: 300px;
}