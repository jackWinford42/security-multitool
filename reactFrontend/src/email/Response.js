
export default function Response({data}) {
  console.debug("RESPONSE COMPONENT")
  if (!data.valid) return <p>For best results, enter a valid email</p>
  return (
    <div className="Response">
      
    </div>
  );
}