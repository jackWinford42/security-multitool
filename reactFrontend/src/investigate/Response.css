#progressBar {
    width: 300px;
}

.responseCard {
    position: relative;
    padding-top: 500px;
    margin: 0 auto;
    width: 700px;
}