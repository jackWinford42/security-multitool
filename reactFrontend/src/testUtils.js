import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };