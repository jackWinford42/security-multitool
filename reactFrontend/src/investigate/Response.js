import "./Response.css";
import { Alert, Card, CardBody} from 'reactstrap';
import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';

export default function Response({data, open}) {
  //console.debug("RESPONSE COMPONENT")

  const percent = 100-data.fraud_score;

  function smtp(score) {
    switch (score) {
      case -1:
        return <p>invalid email address</p>
      case 0:
        return <p>the mail server for this email exists, but it is rejecting all mail</p>
      case 1:
        return <p>the mail server for this email exists, but it is showing a temporary error which is suspicious</p>
      case 2:
        return <p>the mail server exists and accepts all mail</p>
      case 3:
        return <p>there is nothing suspicious about the mail server</p>
      default:
        return <p>invalid email address or server error</p>
    }
  }

  function overall(score) {
    switch (score) {
      case 0:
        return <p>invalid email address</p>
      case 1:
        return <p>the dns for this email is valid, but the mail server is unreachable</p>
      case 2:
        return <p>the dns for this email is valid, but there is a temporary mail rejection error</p>
      case 3:
        return <p>the dns for this email is valid, and accepts all mail</p>
      case 4:
        return <p>there is nothing suspicious about the dns for this email</p>
      default:
        return <p>invalid email address or server error</p>
    }
  }

  if (!data.success || data.overall_score === 0) return <Alert className="redAlert" color="danger">For best results, enter a valid email</Alert>
  return (
    <Card className="Response responseCard">
      <CardBody>
        <h5>This email is {percent}% safe <InfoOutlinedIcon id="infoIcon" onClick={() => open()}/></h5>
        <div className="progress">
          <div id="progressBar" className="progress-bar" role="progressbar" style={{width: `${percent}%`}} aria-valuenow={percent} aria-valuemin="0" aria-valuemax="100"></div>
        </div>
        {overall(data.overall_score)}
        {smtp(data.smtp_score)}
        {data.disposable && <p>This email is suspected of belonging to a temporary or disposable mail service.
          Usually associated with fraudsters and scammers.</p>}
        {data.leaked && <p>This email address is associated with a recent database leak from a third party.
          Leaked accounts pose a risk as they may have become compromised during a database breach.</p>}
        {data.recent_abuse && <p>There has been recent verified abuse with this email address. Abuse could 
          be a confirmed chargeback, fake signup, compromised device, fake app install, or similar malicious 
          behavior within the past few days.</p>}
        {data.honeypot && <p>This email is believed to be a SPAM trap, AKA a honeypot. Bulk mail 
          sent to these emails increases your risk of being blacklisted by large ISPs and ending up in the spam folder.</p>}
      </CardBody>
    </Card>
  );
}
