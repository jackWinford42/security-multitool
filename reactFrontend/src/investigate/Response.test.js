import React from "react";
import { render } from "@testing-library/react";
import Response from "./Response";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Response data={{fraud_score: 10, overall_score: 1, smtp_score: 1, disposable: false, leaked: false, recent_abuse: false, honeypot: false}}/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
      <Response data={{fraud_score: 10, overall_score: 1, smtp_score: 1, disposable: false, leaked: false, recent_abuse: false, honeypot: false}}/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});