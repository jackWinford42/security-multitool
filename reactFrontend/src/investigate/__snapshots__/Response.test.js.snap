// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <p>
    For best results, enter a valid email
  </p>
</DocumentFragment>
`;
